src/firefly/js/visualize/ui/FileUploadViewPanel.jsx

```diff
@@ -72,38 +57,44 @@ const TABLES_ONLY_SUPPORTED_TYPES=[
     FileAnalysisType.Table,
 ];
 
-
-const isTablesOnly= () => getAppOptions()?.uploadPanelLimit==='tablesOnly';
-
-
 const uploadOptions = 'uploadOptions';
 
-let currentAnalysisResult, currentReport, currentSummaryModel, currentDetailsModel;
 const FILE_UPLOAD_KEY= 'file-upload-key-';
 let keyCnt=0;
 
-export function FileUploadViewPanel({setSubmitText}) {
+export function FileUploadViewPanel({setSubmitText, acceptMoc}) {
+
+    const {groupKey}= useContext(FieldGroupCtx);
 
-    const {isLoading,statusKey} = useStoreConnector(() => getComponentState(panelKey, {isLoading:false,statusKey:''}));
+    const {isLoading,statusKey} = useStoreConnector(() => getComponentState(groupKey, {isLoading:false,statusKey:''}));
     const isWsUpdating          = useStoreConnector(() => isAccessWorkspace());
-    const uploadSrc             = useStoreConnector(() => getFieldVal(panelKey, uploadOptions));
-    const {message, analysisResult, report, summaryModel, detailsModel} = useStoreConnector((oldState) => getNextState(oldState));
+    const uploadSrc= useFieldGroupValue(uploadOptions)[0]();
+
+
+    const SUMMARY_TBL_ID = groupKey; //FileUploadAnalysis
+    const DETAILS_TBL_ID = groupKey + '-Details';
+    const UNKNOWN_FORMAT = 'UNKNOWN';
+    const summaryUiId = SUMMARY_TBL_ID + '-UI';
+    const detailsUiId = DETAILS_TBL_ID + '-UI';
+
+    const [getUploadMetaInfo, setUploadMetaInfo]= useFieldGroupMetaState({message:undefined, analysisResult: undefined,
+        report: undefined, summaryModel: undefined, detailsModel: undefined, prevAnalysisResult: undefined, isLoading: undefined}, groupKey);
+
+    let {message, analysisResult, report, summaryModel, detailsModel, prevAnalysisResult} = useStoreConnector(() =>
+        getNextState(groupKey, getUploadMetaInfo()));
```

Comment:
One of the problems with this component is the `getNextState` is very, very difficult to understand. This is largely because it is not a clean function. So I am going to make some extensive suggestion with `getNextState`. The data that is changing should be passed to it and then it creates a new state.

```js
   const {message, analysisResult, report, summaryModel, detailsModel, prevAnalysisResult} =
        useStoreConnector(() => {
            const loadingOp= getLoadingOp();
            const {analysisResult, message}= getField(groupKey, loadingOp) || {};
            const summaryTbl= getTblById(summaryTblId);
            return getNextState(summaryTblId, summaryTbl, detailsTblId, analysisResult, message, getUploadMetaInfo());
        });
```
make changing like this, it is clear what the inputs to getNextState are.